pydatastructs/utils/misc_util.py

```diff
@@ -40,3 +40,30 @@ def __str__(self):
         Used for printing.
         """
         return str((self.left, self.key, self.data, self.right))
+
+
+class GraphNode(object):
+    """
+    Represents node in Graphs.
+
+    Parameters
+    ==========
+
+    data
+        Any valid data to be stored in the node.
+    key
+        Required for comparison operations.
+    """
+
+    __slots__ = ['key', 'data']
+
+    def __new__(cls, key, data):
+        obj = object.__new__(cls)
+        obj.data, obj.key = data, key
+        return obj
+
+    def __str__(self):
+        """
+        Used for printing.
+        """
+        return str((self.key, self.data))
```

Comment:
Is it a good idea to print adjacency list as well? What do you say?

Reply:
Yes can do that (will be better if we let the user know it is an adjacency list), not just print the list as third argument.

Reply:
> will be better if we let the user know it is an adjacency list

How can that be done? Using `dict`?

Reply:
Yes something like `{'adjacent_nodes' : {key1, key2, key3,. , . , . }}`. The value of the `dict` can be a `set`. I think there can be more than 1 edge between 2 nodes right? So when edges are added we want to add the adjacent nodes in the `set`.

Reply:
> I think there can be more than 1 edge between 2 nodes

Can you give an example of such a situation?

I think rather than using `set` we should use `DynamicOneDimensionalArray` as there are many graph algorithms which remove edges, in that case `DynamicOneDimensionalArray` will guarantee amortized constant time.

Reply:
https://en.wikipedia.org/wiki/Multigraph
I was talking about this. If we are only working with simple graphs, then we will always have 1 edge.

Reply:
Well, in the case of multigraphs we may need a create a separate edge set(a multiset) and adjacency node of Graph node is of very little use in that case. Can you share some use cases of multigraphs, like problems from books/papers/university lectures, so that we can think of how concretely we can represent them?

Reply:
http://mathworld.wolfram.com/KoenigsbergBridgeProblem.html 
You can take a look at this. I think for now we can work for simple graphs. Usually multi graph problems are related to those with weighted edges mostly. Like finding shortest paths etc.

Reply:
Multigraphs may need a different discussion. The current structure of `GraphNode` will be compatible with that. We can keep adjacency list empty if needed in multigraphs. 
Can you raise an issue for crearting MultiGraphs? We can work on it after graphs are added.